src/PrestaShopBundle/Controller/Admin/Improve/Design/MailThemeController.php

```diff
@@ -0,0 +1,364 @@
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Controller\Admin\Improve\Design;
+
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShop\PrestaShop\Core\Exception\FileNotFoundException;
+use PrestaShop\PrestaShop\Core\Exception\InvalidArgumentException;
+use PrestaShop\PrestaShop\Core\Form\FormHandlerInterface;
+use PrestaShop\PrestaShop\Core\Language\LanguageInterface;
+use PrestaShop\PrestaShop\Core\Language\LanguageRepositoryInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\Layout\LayoutInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\MailTemplateInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\MailTemplateRendererInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\ThemeCatalogInterface;
+use PrestaShop\PrestaShop\Core\MailTemplate\ThemeInterface;
+use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
+use PrestaShopBundle\Form\Admin\Improve\Design\MailTheme\GenerateMailsType;
+use PrestaShopBundle\Security\Annotation\AdminSecurity;
+use PrestaShopBundle\Service\MailTheme\MailThemeGenerator;
+use Symfony\Component\Form\Form;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+/**
+ * Class MailThemeController manages mail theme generation, you can define the shop
+ * mail theme, and regenerate mail in a specific language.
+ *
+ * Accessible via "Design > Mail Theme"
+ */
+class MailThemeController extends FrameworkBundleAdminController
+{
+    /**
+     * Show mail theme settings and generation page.
+     *
+     * @AdminSecurity("is_granted('read', request.get('_legacy_controller'))")
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function indexAction(Request $request)
+    {
+        $legacyController = $request->attributes->get('_legacy_controller');
+        $generateThemeMailsForm = $this->createForm(GenerateMailsType::class);
+        /** @var ThemeCatalogInterface $themeCatalog */
+        $themeCatalog = $this->get('prestashop.core.mail_template.theme_catalog');
+        $mailThemes = $themeCatalog->listThemes();
+
+        return $this->render('@PrestaShop/Admin/Improve/Design/MailTheme/index.html.twig', [
+            'layoutHeaderToolbarBtn' => [],
+            'layoutTitle' => $this->trans('Mail Theme', 'Admin.Navigation.Menu'),
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink($legacyController),
+            'mailThemeConfigurationForm' => $this->getMailThemeFormHandler()->getForm()->createView(),
+            'generateMailsForm' => $generateThemeMailsForm->createView(),
+            'mailThemes' => $mailThemes,
+        ]);
+    }
+
+    /**
+     * Manage generation form post and generate mails.
+     *
+     * @AdminSecurity("is_granted(['create', 'update'], request.get('_legacy_controller'))")
```

Comment:
```suggestion
     * @AdminSecurity("is_granted(['create'], request.get('_legacy_controller'))")
```
ping @matks

Reply:
Does this mean you need both permissions? Or one of them?
Because as you don't know if the mail exist or not, depending on the context this action can be considered `create` or `update` action

Reply:
As I remember, him and Mickael said me there must be one action for create, and one for update :thinking: Or I'm maybe wrong :P

Reply:
Done